orienteer-graph/src/main/java/org/orienteer/graph/component/widget/GraphVerticesWidget.java

```diff
@@ -26,51 +26,57 @@
-
-		@Override
-		public String apply(ODocument vertex) {
-			Object fieldIn = getModelObject().field("in");
-            String direction;
-			if (fieldIn != null) {
-                direction = ((OIdentifiable)fieldIn).getIdentity().equals(vertex.getIdentity()) ? "in":"out";
-            } else direction = "empty";
-            return getLocalizer().getString("widget.document.vertices.title." + direction, GraphVerticesWidget.this);
-		}
-		
-	}
 
     @Inject
     private OClassIntrospector oClassIntrospector;
 
     @SuppressWarnings("unchecked")
     public GraphVerticesWidget(String id, final IModel<ODocument> model, IModel<ODocument> widgetDocumentModel) {
         super(id, model, widgetDocumentModel);
+    }
 
-        IModel<DisplayMode> modeModel = DisplayMode.VIEW.asModel();
-        OProperty nameProperty = oClassIntrospector.getNameProperty(getModelObject().getSchemaClass());
-        OEntityColumn entityColumn = new OEntityColumn(nameProperty, true, modeModel);
-
-        ODocumentDescriptionColumn directionColumn = new ODocumentDescriptionColumn(
-                new StringResourceModel("property.direction", this, Model.of()),
-                new DirectionLocalizer());
-
-        OQueryDataProvider<ODocument> provider = new OQueryDataProvider<>("select expand(bothV()) from " + model.getObject().getIdentity());
+    @Override
+    protected void onInitialize() {
+        super.onInitialize();
+        String sql = "select expand(bothV()) from " + getModelObject().getIdentity();
         GenericTablePanel<ODocument> tablePanel = new GenericTablePanel<>("vertices",
-                Lists.newArrayList(entityColumn, directionColumn),
-                provider,//setParameter does not work here
-                2);
+                createColumns(),
+                new OQueryDataProvider<>(sql), //setParameter does not work here
+                2
+        );
         OrienteerDataTable<ODocument, String> table = tablePanel.getDataTable();
-        table.addCommand(new ExportCommand<>(table, new StringResourceModel("export.filename.vertices", new ODocumentNameModel(model))));
+        table.addCommand(new ExportCommand<>(table, new StringResourceModel("export.filename.vertices", new ODocumentNameModel(getModel()))));
+
         add(tablePanel);
-        add(DisableIfDocumentNotSavedBehavior.INSTANCE,UpdateOnActionPerformedEventBehavior.INSTANCE_ALL_CONTINUE);
+        add(DisableIfDocumentNotSavedBehavior.INSTANCE, UpdateOnActionPerformedEventBehavior.INSTANCE_ALL_CONTINUE);
+    }
+
+    private List<IColumn<ODocument, String>> createColumns() {
+        List<IColumn<ODocument, String>> columns = new LinkedList<>();
+        columns.add(createNameColumn());
+        columns.add(createDescriptionColumn());
+        return columns;
+    }
+
+    private IColumn<ODocument, String> createNameColumn() {
+        ODocument edge = getModelObject();
+        ODocument in = edge.field("in");
```

Comment:
It's taking only in. More accurate will be to take common parent by method probeOClass